horace_core/sqw/page_operations/PageOp_unary_op.m

```diff
@@ -0,0 +1,53 @@
+classdef PageOp_unary_op < PageOpBase
+    % Single page pixel operation used by
+    % unary operation manager
+    %
+    %
+    properties
+        % property contains
+        op_handle;
+    end
+
+    methods
+        function obj = PageOp_unary_op(varargin)
+            obj = obj@PageOpBase(varargin{:});
+
+        end
+
+        function [obj,in_obj] = init(obj,in_obj,operation)
+            [obj,in_obj] = init@PageOpBase(obj,in_obj);
+            obj.op_handle = operation;
+            obj.op_name_ = sprintf('unary op: %s',func2str(operation));
+            if ~obj.changes_pix_only
+                obj.var_acc_ = zeros(numel(obj.npix),1);
+            end
+        end
+
+        function obj = apply_op(obj,npix_block,npix_idx)
+            signal = obj.page_data_(obj.signal_idx_,:);
+            var    = obj.page_data_(obj.var_idx_,:);
+            pg_result = obj.op_handle(sigvar(signal, var));
+
+            signal = pg_result.s;
+            var    = pg_result.e;
```

Comment:
any particular reason for using these intermediate variables rather than assigning result directly in the next 2 lines?